index.ts

```diff
@@ -200,21 +204,50 @@ function suiteClassCallback(target: SuiteCtor, context: TestFunctions) {
 				testName = testName || (<any>method).name;
 				if (shouldPending && !shouldSkip && !shouldOnly) {
 					context.it.skip(testName);
-				} else if (method.length > 0) {
-					testFunc(testName, noname(function(this: Mocha.ITestCallbackContext, done) {
-						applyDecorators(this, prototype, method, instance);
-						applyTestTraits(this, instance, method);
-						return method.call(instance, done);
-					}));
-				} else {
-					testFunc(testName, noname(function(this: Mocha.ITestCallbackContext) {
-						applyDecorators(this, prototype, method, instance);
-						applyTestTraits(this, instance, method);
-						return method.call(instance);
-					}));
+				} else if (parameters) {
+					parameters.forEach((parameterOptions, i) => {
+						const [testFunc, mark, name, parameter] = parameterOptions;
+
+						let parametersTestName = `${testName}_${i}`;
+						if (name) {
+							parametersTestName = name;
+						} else if (nameForParameters) {
+							parametersTestName = nameForParameters(parameter);
+						}
+
+						shouldSkip = mark === skipSymbol;
+						shouldOnly = mark === onlySymbol;
+						shouldPending = mark === pendingSymbol;
+
+						if (shouldPending && !shouldSkip && !shouldOnly) {
+							context.it.skip(testName);
+						}
+
+						applyTestFunc(testFunc, parametersTestName, method, [parameter], method.length <= 1);
+                    });
+                } else {
```

Comment:
Maybe white-space is skewed here. I hate to return PRs for such non-productive stuff so I'd be ok to merge it. But we should add tslint and autofix scripts in future. For reference: https://github.com/NativeScript/css-parser/blob/1f727144b8b6dfe178f63f4876183bb67976641a/package.json#L10